### PR TITLE
Feature/cli rework

### PR DESCRIPTION
**Pull request by mcrowson**
## Description

Redid the CLI so we could have help on the commands. A few questions:
1. The test_cli_args currently just returns and prints "The command is not recognized" The rework causes this to fail because derp was never a command. If the intention was to exit, i have modified the test to grab that exit. If the intent was to succeed, let me know what success looks like and I can update the test so it doesn't exit.
2. Do both -s and -a arguments come before the command argument?
3. Do both -y and --all arguments come after the command argument?
## GitHub Issues

#404

### CODE REVIEW

**Comment by mcrowson** [comment]
Looks like it has issues with how we're getting the versions. This might be easier to merge after https://github.com/Miserlou/Zappa/pull/426 or implement a different version method.

**Comment by mcrowson** [comment]
It looks like there is interest in progressing the CLI, If this PR seems more appealing than #519 I can bring it up to date with master.

File: zappa/cli.py [hunk 1]
```diff
@@ -132,49 +117,84 @@ def handle(self, argv=None):
         Parses command, load settings and dispatches accordingly.
 
         """
-        help_message = "Please supply a command to execute. Can be one of: {}".format(', '.join(x for x in sorted(CLI_COMMANDS)))
-
-        parser = argparse.ArgumentParser(description='Zappa - Deploy Python applications to AWS Lambda and API Gateway.\n')
-        parser.add_argument('command_env', metavar='U', type=str, nargs='*', help=help_message)
-        parser.add_argument('-n', '--num-rollback', type=int, default=0,
-                            help='The number of versions to rollback.')
+        parser = argparse.ArgumentParser(description='Zappa - Deploy Python applications to AWS Lambda and API Gateway.\n', version=pkg_resources.require("zappa")[0].version, prog="zappa")
```
Comment (rmoe) [inline zappa/cli.py]: Neat, I've never seen version= used here to provide a version argument. TIL.

Comment (mcrowson) [inline zappa/cli.py]: yea, argparse has version built in. https://docs.python.org/2.7/library/argparse.html#action

**Comment by mcrowson** [comment]
@rmoe is taking over this stuff with #519

**Pull request closed without merging**